pkg/cid/cid.go

```diff
@@ -65,26 +74,26 @@ func GetX509Certificate(stub ChaincodeStubInterface) (*x509.Certificate, error)
 	return c.GetX509Certificate()
 }
 
-// ClientIdentityImpl implements the ClientIdentity interface
-type clientIdentityImpl struct {
+// ClientID holds the information of the transaction creator.
+type ClientID struct {
 	stub  ChaincodeStubInterface
 	mspID string
 	cert  *x509.Certificate
 	attrs *attrmgr.Attributes
 }
 
-// New returns an instance of ClientIdentity
-func New(stub ChaincodeStubInterface) (ClientIdentity, error) {
-	c := &clientIdentityImpl{stub: stub}
+// New returns an instance of ClientID
+func New(stub ChaincodeStubInterface) (ClientID, error) {
+	c := ClientID{stub: stub}
 	err := c.init()
```

Comment:
I'm surprised we didn't see test failures here since this is now a value object instead of reference.